git_machete/client.py

```diff
@@ -2168,17 +2168,16 @@ def __get_added_remote_name_or_none(self, github_domain: str, remote_url: str) -
                 return remote
         return None
 
-    def retarget_github_pr(self, head: LocalBranchShortName) -> None:
+    def retarget_github_pr(self, head: LocalBranchShortName, ignore_if_missing: bool) -> None:
         domain = self.__derive_github_domain()
         remote_org_repo = self.__derive_remote_and_github_org_and_repo(domain=domain, branch_used_for_tracking_data=head)
         github_client = GitHubClient(domain=domain, organization=remote_org_repo.organization, repository=remote_org_repo.repository)
 
         debug(f'organization is {remote_org_repo.organization}, repository is {remote_org_repo.repository}')
 
-        pr: Optional[GitHubPullRequest] = github_client.derive_pull_request_by_head(head)
+        pr: Optional[GitHubPullRequest] = github_client.derive_pull_request_by_head(head, ignore_if_missing)
```

Comment:
Hmm I'd rather **not** pollute `GitHubClient` with details like `ignore_if_missing`... let's throw an exception in `GitHubClient.derive_pull_request_by_head`, and handle it here in `retarget_github_pr` (possibly ignoring it, if `ignore_if_missing` is True)